token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -700,9 +723,20 @@ fn process_destination_for_transfer(
         return Err(TokenError::ConfidentialTransferElGamalPubkeyMismatch.into());
     }
 
-    let new_destination_pending_balance = ops::add_with_lo_hi(
-        &destination_confidential_transfer_account.pending_balance,
+    if destination_confidential_transfer_account.pending_balance_credit_counter
+        == destination_confidential_transfer_account.maximum_pending_balance_credit_counter
+    {
+        return Err(TokenError::MaximumPendingBalanceCreditCounterExceeded.into());
+    }
+
+    let new_destination_pending_balance_lo = ops::add(
+        &destination_confidential_transfer_account.pending_balance_lo,
         destination_ciphertext_lo,
+    )
+    .ok_or(ProgramError::InvalidInstructionData)?;
+
+    let new_destination_pending_balance_hi = ops::add(
+        &destination_confidential_transfer_account.pending_balance_hi,
```

Comment:
This may be a silly question, but since you only do the validation that the transfer amount fits in 48 bytes on the client, how are you sure that it fits over here on the program side?

Reply:
Yep that's a valid question. The range is checked via zkp's. The proof program will check the validity of the range proof that is attached with a transfer instruction, which will guarantee that the transfer amount is 48 bits. This update to zkp generation should already be checked into the zk-token-sdk.